app/models/agents/read_file_agent.rb

```diff
@@ -0,0 +1,49 @@
+module Agents
+  class ReadFileAgent < Agent
+    include FormConfigurable
+    include FileHandling
+
+    cannot_be_scheduled!
+
+    def default_options
+      {
+        'data_key' => 'data'
+      }
+    end
+
+    description do
+      <<-MD
+        The ReadFileAgent takes events from `FileHandling` agents, reads the file, and emits the contents as a string.
```

Comment:
Yes, I don't know a way to do streaming writes into mysql or postgresql. But the same applies for the WebsiteAgent, you can upload a huge HTML or XML file somewhere and let Huginn check it every minute.